src/lib.rs

```diff
@@ -307,19 +311,17 @@ mod test {
         let wallet_configs =
             hashmap![sender_index => sender_balance, recipient_index => recipient_balance];
         let (_, mut actors) = get_network(group_count, replica_count, wallet_configs);
-        let mut sender = actors
-            .remove(&sender_index)
-            .ok_or_else(|| Error::Unexpected("Sender missing from actors".to_string()))?;
+        let mut sender = actors.remove(&sender_index).ok_or(Error::MissingSender)?;
```

Comment:
This ok.or pattern seems strange here. Usually, it is to set another value? I think it should just be a try `?` Same in a few other places.

Reply:
Looking through them, seems like they're mostly there for increased info on test failures.

This line for example, will return `None` if no balance is found, but would just error out with an unrwap_none error no?  (and this whole block after ln 181 is in tests only.

Reply:
If only tests then cool, however the pattern does seem weird mixing Option and Result/Error. We should look into this soon.